management/clipper_manager.py

```diff
@@ -831,21 +849,98 @@ def add_container(self, model_name, model_version):
                 add_container_cmd = (
                     "docker run -d --network={nw} -v {path}:/model:ro "
                     "-e \"CLIPPER_MODEL_NAME={mn}\" -e \"CLIPPER_MODEL_VERSION={mv}\" "
-                    "-e \"CLIPPER_IP=query_frontend\" -e \"CLIPPER_INPUT_TYPE={mip}\" "
+                    "-e \"CLIPPER_IP=query_frontend\" -e \"CLIPPER_INPUT_TYPE={mip}\" -l \"{clipper_label}\" "
                     "{image}".format(
                         path=model_data_path,
                         nw=DOCKER_NW,
                         image=image_name,
                         mn=model_name,
                         mv=model_version,
-                        mip=model_input_type))
+                        mip=model_input_type,
+                        clipper_label=CLIPPER_DOCKER_LABEL))
                 result = self._execute_root(add_container_cmd)
                 return result.return_code == 0
             else:
                 print("Cannot start containers for externally managed model %s"
                       % model_name)
                 return True
 
+    def get_clipper_logs(self, additional_image_names=None):
+        """Returns true if the provided conda environment is compatible with the container os.
+
+        If packages listed in specified conda environment file have conflicting dependencies,
+        this function will warn the user and return False. If packages don't exist in the
+        container's conda channel, this function will warn the user and remove those packages.
+
+        Parameters
+        ----------
+        additional_image_names : list(str), optional
+            Any additional Docker container image names whose logs should
+            be retrieved. This should only be necessary if you are using model containers
+            that don't use a standard clipper RPC container (clipper/py-rpc, clipper/java-rpc)
+            as a base image.
+
+        Returns
+        -------
+        list(str)
+            Returns a list of local filenames containing the Docker container log snapshots.
+        """
+        container_ids = self._get_clipper_container_ids()
+        if additional_image_names is not None:
+            for image in additional_image_names:
+                container_ids.extend(
+                    self._get_docker_container_id_from_image(image))
+        try:
+            os.mkdir(CLIPPER_LOGS_PATH)
+        except OSError:
+            pass
```

Comment:
Why do we pass here? If we failed to create the directory we should not continue execution. It makes sense to pass only if the command failed due to a preexisting directory.